src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -411,7 +411,17 @@ public virtual Task<bool> SetStepFiltering(bool enabled)
 
         private StringBuilder BuildBreakInsert(string condition)
         {
-            StringBuilder cmd = new StringBuilder("-break-insert -f ");
+            string pendingFlag = "-f ";
+            if(this.Mode == MIMode.Lldb)
```

Comment:
Why not make this method `protected` and override it in `LlldbMICommandFactory`?